datafusion/core/src/datasource/physical_plan/parquet.rs

```diff
@@ -1921,6 +1861,103 @@ mod tests {
         assert_eq!(predicate.unwrap().to_string(), filter_phys.to_string());
     }
 
+    /// Empty file won't get partitioned
+    #[tokio::test]
+    async fn parquet_exec_repartition_empty_file_only() {
+        let partitioned_file_empty = PartitionedFile::new("empty".to_string(), 0);
+        let file_group = vec![vec![partitioned_file_empty]];
+
+        let parquet_exec = ParquetExec::new(
+            FileScanConfig {
+                object_store_url: ObjectStoreUrl::local_filesystem(),
+                file_groups: file_group,
+                file_schema: Arc::new(Schema::empty()),
+                statistics: Statistics::default(),
+                projection: None,
+                limit: None,
+                table_partition_cols: vec![],
+                output_ordering: vec![],
+                infinite_source: false,
+            },
+            None,
+            None,
+        );
+
+        let partitioned_file = parquet_exec
+            .get_repartitioned(4, 0)
+            .base_config()
+            .file_groups
+            .clone();
+
+        assert!(partitioned_file[0][0].range.is_none());
+    }
+
+    // Repartition when there is a empty file in file groups
+    #[tokio::test]
+    async fn parquet_exec_repartition_empty_files() {
```

Comment:
maybe as a follow on the tests for file repartitioning could be moved to datafusion/core/src/datasource/physical_plan/mod.rs to be in the same module they are testing (as they are now no longer specific to parquet)